packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,286 @@
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input />
+    </va-form>
+    <va-button @click="$refs.form.focus()">
+      Focus first input
+    </va-button>
+  `,
+})
+
+Focus.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: '' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus first input' }) as HTMLElement
+
+  await step('Focusses first input', async () => {
+    await userEvent.click(button)
+    expect(firstInput).toHaveFocus()
+  })
+}
+
+export const FocusInvalid = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input error />
+    </va-form>
+    <va-button @click="$refs.form.focusInvalidField()">
+      Focus invalid
+    </va-button>
+  `,
+})
+
+FocusInvalid.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const input = canvas.getByRole('textbox', { name: '' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus invalid' }) as HTMLElement
+
+  await step('Focusses invalid input', async () => {
+    await userEvent.click(button)
+    expect(input).toHaveFocus()
+  })
+}
+
+export const ValidateAndResetValidation = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'do not reset' }),
+  template: `
+    <va-form ref="form">
+      <va-input v-model="input"/>
+      <va-input :rules="[() => 'error']"/>
+    </va-form>
+    <va-button @click="$refs.form.validate()">
+      Validate
+    </va-button>
+    <va-button @click="$refs.form.resetValidation()">
+      Reset validation
+    </va-button>
+  `,
+})
+
+ValidateAndResetValidation.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const [firstInput, secondInput] = canvas.getAllByRole('textbox', { name: '' }) as HTMLElement[]
+  const validateButton = canvas.getByRole('button', { name: 'Validate' }) as HTMLElement
+  const resetButton = canvas.getByRole('button', { name: 'Reset validation' }) as HTMLElement
+
+  await step('Validates input vith error', async () => {
+    await userEvent.click(validateButton)
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('true')
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+
+  await step('Reset inputs validation', async () => {
+    await userEvent.click(resetButton)
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('false')
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+}
+
+export const Reset = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ username: 'reset' }),
+  template: `
+    <va-form
+      ref="form"
+      immediate
+    >
+      <va-input v-model="username"/>
+      <va-input :rules="[() => 'error']"/>
+    </va-form>
+    <va-button @click.prevent="$refs.form.reset()">
```

Comment:
We need 2 buttons in total:
* `Fill` - (set value and run validation)
* `Reset` - refs.form.reset

Right now - it's not possible to interact manually.